tests/integrations/raftstore/test_merge.rs

```diff
@@ -938,7 +938,10 @@ fn test_merge_isolated_store_with_no_target_peer() {
     let right_on_store3 = find_peer(&right, 3).unwrap().to_owned();
     pd_client.must_remove_peer(right.get_id(), right_on_store3);
 
+    // Ensure snapshot is sent and applied.
+    must_get_equal(&cluster.get_engine(4), b"k4", b"v1");
```

Comment:
`must_get_equal(&cluster.get_engine(4), b"k22", b"v22");` will also ensure that

Reply:
They are different. `k22` can only ensure snapshot is applied. Additional check ensure snapshot is applied before proposing `put(k22, v22)`, so follower's `matched` must be greater than 0.